common/utils/yaml_test.go

```diff
@@ -80,10 +79,42 @@ func TestMergeYAMLWithSlices(t *testing.T) {
 
 			var gotYAML map[string]interface{}
 			var expectedYAML map[string]interface{}
-
-			gotMap := yaml.Unmarshal([]byte(got), &gotYAML)
-			expectedMap := yaml.Unmarshal([]byte(tc.expectedResult), &expectedYAML)
-			assert.Equal(t, gotMap, expectedMap, "expected %v, got %v", expectedMap, gotMap)
+			err = yaml.Unmarshal([]byte(got), &gotYAML)
+			require.NoError(t, err, "unexpected error: %v", err)
+			err = yaml.Unmarshal([]byte(tc.expectedResult), &expectedYAML)
+			require.NoError(t, err, "unexpected error: %v", err)
+			assertMapsEqual(t, expectedYAML, gotYAML)
 		})
 	}
 }
+
+func assertMapsEqual(t *testing.T, expected, actual map[string]interface{}) {
+	assert.Equal(t, len(expected), len(actual), "expected %v, got %v", expected, actual)
+	for k, v := range expected {
+		switch v.(type) {
+		case map[string]interface{}:
+			assertMapsEqual(t, v.(map[string]interface{}), actual[k].(map[string]interface{}))
+		case []interface{}:
+			assertSlicesEqual(t, v.([]interface{}), actual[k].([]interface{}))
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
	assert.ElementsMatch(t, v.([]interface{}), actual[k].([]interface{}))
```
should work https://pkg.go.dev/github.com/stretchr/testify/assert#Assertions.ElementsMatch

Reply:
I'll take that one!